plugin/path_login.go

```diff
@@ -320,9 +320,6 @@ func (b *GcpAuthBackend) pathIamLogin(ctx context.Context, req *logical.Request,
 	if err != nil {
 		return logical.ErrorResponse("unable to retrieve GCP configuration"), nil
 	}
-	if conf == nil {
```

Comment:
For this location, the effect of the config being unset but populated is basically no change from before we did any work on this plugin. Before, the config was mainly used in building the client, and if the client had no available credentials, it would fail there. It still will with this approach.